packages/core/src/components/TextField/getMaskedValue.ts

```diff
@@ -1,29 +1,23 @@
-import type { ChangeEvent } from 'react';
-
-const getMaskedValue = (event: ChangeEvent<HTMLInputElement>, mask: string): string => {
-    const { value } = event.target,
+const applyMasking = (value: string, mask: string): string => {
+    const { length } = value,
+        lastChar = value[length - 1],
```

Comment:
Instead of using index notation. We can use `charAt` since it also covers some edge cases and has better compatibility overall

Source: https://stackoverflow.com/questions/5943726/string-charatx-or-stringx

Reply:
Done. Nice finding. :)